.changelog/23646.txt

```diff
@@ -0,0 +1,3 @@
+```release-note:new-data-source
```

Comment:
The name of the changelog file should be a PR number, so `24423.txt` in this case.
https://github.com/hashicorp/terraform-provider-aws/blob/main/docs/contributing/pullrequest-submission-and-lifecycle.md#changelog-format